File: src/main/java/com/ideabaker/samples/springsecurity/model/Movie.java
package com.ideabaker.samples.springsecurity.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@ToString
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "movies")
public class Movie {

  @Id
  private Long id;
  private String title;
  private String genre;

}

File: src/main/kotlin/com.ideabaker.samples.springsecurity/domain/Movie.kt
package com.ideabaker.samples.springsecurity.domain

import org.springframework.data.mongodb.core.mapping.Document
import javax.validation.constraints.NotBlank

/**
 *
 * @author Arthur Kazemi<bidadh@gmail.com>
 * @since 2018-12-29 10:47
 */
@Document(collection = "movies")
class Movie(
        @NotBlank(message = "Movie title cannot be empty") val title: String,
        val genre: String
): AbstractDocument()
